Source/TestDafny/MultiBackendTest.cs

```diff
@@ -207,6 +207,17 @@ int ExpectedExitCode
         if (result != 0) {
           success = false;
         }
+
+        if (compiler.TargetId == "cs") {
+          // C# is a bit unusual in that the runtime behaves a little differently
+          // depending on whether it is included as source or referenced as DafnyRuntime.dll
+          // (because of "#ifdef ISDAFNYRUNTIMELIB" directives).
```

Comment:
I didn't get what these directives are for. Could you add documentation on that somewhere close to the C# runtime or compiler?